File: SDKs/amazonaws.com/appmesh/2019-01-25/java/lib/src/main/java/openapisdk/models/shared/TlsValidationContext.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TlsValidationContext
 * An object that represents how the proxy will validate its peer during Transport Layer Security (TLS) negotiation.
**/
public class TlsValidationContext {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("subjectAlternativeNames")
    public SubjectAlternativeNames subjectAlternativeNames;
    public TlsValidationContext withSubjectAlternativeNames(SubjectAlternativeNames subjectAlternativeNames) {
        this.subjectAlternativeNames = subjectAlternativeNames;
        return this;
    }
    @JsonProperty("trust")
    public TlsValidationContextTrust trust;
    public TlsValidationContext withTrust(TlsValidationContextTrust trust) {
        this.trust = trust;
        return this;
    }
}

File: SDKs/amazonaws.com/appmesh/2019-01-25/python/src/sdk/models/shared/tlsvalidationcontext.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import subjectalternativenames as shared_subjectalternativenames
from ..shared import tlsvalidationcontexttrust as shared_tlsvalidationcontexttrust


@dataclass_json
@dataclasses.dataclass
class TLSValidationContext:
    r"""TLSValidationContext
    An object that represents how the proxy will validate its peer during Transport Layer Security (TLS) negotiation.
    """
    
    trust: shared_tlsvalidationcontexttrust.TLSValidationContextTrust = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('trust') }})
    subject_alternative_names: Optional[shared_subjectalternativenames.SubjectAlternativeNames] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subjectAlternativeNames') }})
    
